web/satellite/src/components/accessGrants/AccessGrantsItem2.vue

```diff
@@ -33,31 +40,52 @@
 
 <script lang="ts">
 import { Component, Prop, Vue } from 'vue-property-decorator';
-
+import TrashIconWhite from '@/../static/images/accessGrants/trashIcon.svg';
+import TrashIconBlack from '@/../static/images/accessGrants/trashIcon-black.svg';
 import { AccessGrant } from '@/types/accessGrants';
 
+import { ACCESS_GRANTS_ACTIONS } from '@/store/modules/accessGrants';
+
+const {
+    TOGGLE_SELECTION,
+} = ACCESS_GRANTS_ACTIONS;
+
 // @vue/component
 @Component({
     components: {
+        TrashIconWhite,
+        TrashIconBlack,
     },
 })
 export default class AccessGrantsItem extends Vue {
     @Prop({ default: new AccessGrant('', '', new Date(), '') })
     private readonly itemData: AccessGrant;
     private popupVisible = false;
+    private isPopupHovered = false;
 
     public togglePopupVisibility(): void {
         this.popupVisible = !this.popupVisible;
     }
+
+    /**
+     * Toggles access grant selection.
+     */
+    public async toggleSelection(): Promise<void> {
+        await this.$store.dispatch(TOGGLE_SELECTION, this.itemData);
+        this.$emit('altMethod');
+        this.togglePopupVisibility();
+        this.isPopupHovered = false;
+
+    }
 }
 </script>
 
 <style scoped lang="scss">
     @mixin popup-menu-button {
         padding: 0 15px;
-        height: 50%;
-        line-height: 55px;
-        text-align: left;
+        height: 100%;
```

Comment:
since we have vertical padding I think we should set this value like that
`calc(100% - 30px)`

Reply:
Oh, I messed up. `height: 100%` is ok.
`padding: 0 15px;` above is horizontal padding not vertical. 
Ignore this comment please